BlueprintUI/Sources/Element/ElementContent.swift

```diff
@@ -98,27 +89,41 @@ extension ElementContent {
         init(layout: LayoutType) {
             self.layout = layout
         }
+        
+        /// Adds the given child element.
+        public mutating func add(element: Element, traits: LayoutType.Traits = LayoutType.defaultTraits, key: AnyHashable? = nil) {
+            let child = Child(
+                element: element,
+                content: element.content,
+                traits: traits,
+                key: key
+            )
+            
+            children.append(child)
+        }
+        
+        fileprivate struct Child : Measurable {
 
-    }
-
+            var element: Element
+            var content: ElementContent
+            var traits: LayoutType.Traits
+            var key: AnyHashable?
 
+            func measure(in constraint: SizeConstraint) -> CGSize {
+                content.measure(in: constraint)
+            }
+        }
+    }
 }
 
-extension ElementContent.Builder {
-
-    /// Adds the given child element.
-    public mutating func add(traits: LayoutType.Traits = LayoutType.defaultTraits, key: AnyHashable? = nil, element: Element) {
-        let child = Child(
-            traits: traits,
-            key: key,
-            content: element.content,
-            element: element)
-        children.append(child)
-    }
 
+fileprivate protocol AnyContentStorage : Measurable {
```

Comment:
Usually when I see a type prefixed with `Any`, it's a type-erasing struct wrapper, like [AnyIterator, AnySequence, AnyHashable, etc](https://developer.apple.com/documentation/swift/swift_standard_library/collections/supporting_types). I would prefer to keep the protocol as-is and give the implementation a more expressive name.

Reply:
I took to this pattern after using it for type erasure in Listable, since i liked that it meant you didnt need to define yet another wrapper type to do erasure. I don't think the actual type of the Any* thing doesnt really matter, IMO

Reply:
I kind of agree that it makes more sense not to treat this as erasure when it's really just the base functionality for building storage. Maybe one of the names @watt suggested below makes sense? Not a super strong opinion, after all this is kind of like `AnyObject` right? (Types conform to it, aren't wrapped it).

Reply:
It's too bad objects and protocols cant have the same name like they do in ObjC